app/views/handlers/individual_response.py

```diff
@@ -139,12 +143,41 @@ def get_context(self):
         return build_question_context(self.rendered_block, self.form)
 
     def _publish_fulfilment_request(self, mobile_number=None):
+        self._check_individual_response_count()
         topic_id = current_app.config["EQ_FULFILMENT_TOPIC_ID"]
         fulfilment_request = FulfilmentRequest(self._metadata, mobile_number)
         return current_app.eq["publisher"].publish(
             topic_id, message=fulfilment_request.payload
         )
 
+    def _check_individual_response_count(self):
+        if (
+            self._questionnaire_store.response_metadata.get(
+                "individual_response_count", 0
+            )
+            >= current_app.config["EQ_INDIVIDUAL_RESPONSE_LIMIT"]
+        ):
+            raise IndividualResponseLimitExceeded(
+                "Individual response limit has been exceeded"
+            )
+
+    def _update_individual_response_count(self):
+        response_metadata = self._questionnaire_store.response_metadata
+
+        if response_metadata.get("individual_response_count"):
+            response_metadata["individual_response_count"] += 1
+        else:
+            response_metadata["individual_response_count"] = 1
+
+    def _update_questionnaire_store_on_publish(self):
+        self._update_section_status(CompletionStatus.INDIVIDUAL_RESPONSE_REQUESTED)
+        self._update_individual_response_count()
+        self._save_questionnaire_store()
```

Comment:
Calling this makes the saving conditional on the section status being updated, but if we are updating the individual response count, we want to be sure we are saving. It might be better to call the save explicitly here; and remove the  `_save_questionnaire_store` method as there would only be one other method that calls it.